src/main/java/net/minecraftforge/fmp/block/BlockCoverable.java

```diff
@@ -0,0 +1,730 @@
+        RayTraceResult result = tile != null ? tile.getMicroblockContainer().getPartContainer().collisionRayTrace(start, end)
+                : null;
+        RayTraceResult hit = collisionRayTraceDefault(state, world, pos, start, end);
+        if (result == null || (hit != null && hit.hitVec.squareDistanceTo(start) < result.hitVec.squareDistanceTo(start)))
+        {
+            bounds = getSelectedBoundingBoxDefault(state, world, pos).offset(-pos.getX(), -pos.getY(), -pos.getZ());
+            return hit;
+        }
+        bounds = result.boxHit;
+        return result;
+    }
+
+    public RayTraceResult collisionRayTraceDefault(IBlockState state, World world, BlockPos pos, Vec3d start, Vec3d end)
+    {
+        return super.collisionRayTrace(state, world, pos, start, end);
+    }
+
+    @Override
+    public final AxisAlignedBB getSelectedBoundingBox(IBlockState state, World worldIn, BlockPos pos)
+    {
+        return bounds.offset(pos);
+    }
+
+    public AxisAlignedBB getSelectedBoundingBoxDefault(IBlockState state, World worldIn, BlockPos pos)
+    {
+        return super.getSelectedBoundingBox(state, worldIn, pos);
+    }
+
+    @Override
+    public final void addCollisionBoxToList(IBlockState state, World world, BlockPos pos, AxisAlignedBB entityBox,
+            List<AxisAlignedBB> collidingBoxes, Entity collidingEntity)
+    {
+        addCollisionBoxToListDefault(state, world, pos, entityBox, collidingBoxes, collidingEntity);
+        IMicroblockContainerTile tile = getMicroblockTile(world, pos);
+
+        List<AxisAlignedBB> list = new ArrayList<AxisAlignedBB>();
+        AxisAlignedBB box = entityBox.offset(-pos.getX(), -pos.getY(), -pos.getZ());
+        tile.getMicroblockContainer().getPartContainer().addCollisionBoxes(box, list, collidingEntity);
+        for (AxisAlignedBB aabb : list)
+        {
+            collidingBoxes.add(aabb.offset(pos));
+        }
+    }
+
+    public void addCollisionBoxToListDefault(IBlockState state, World worldIn, BlockPos pos, AxisAlignedBB entityBox,
+            List<AxisAlignedBB> collidingBoxes, Entity entityIn)
+    {
+        super.addCollisionBoxToList(state, worldIn, pos, entityBox, collidingBoxes, entityIn);
+    }
+
+    @Override
+    public final int getLightValue(IBlockState state, IBlockAccess world, BlockPos pos)
+    {
+        IMicroblockContainerTile tile = getMicroblockTile(world, pos);
+        return Math.max(tile != null ? tile.getMicroblockContainer().getPartContainer().getLightValue() : 0,
+                getLightValueDefault(state, world, pos));
+    }
+
+    public int getLightValueDefault(IBlockState state, IBlockAccess world, BlockPos pos)
+    {
+        return super.getLightValue(state, world, pos);
+    }
+
+    @Override
+    public final ItemStack getPickBlock(IBlockState state, RayTraceResult target, World world, BlockPos pos, EntityPlayer player)
+    {
+        if (target instanceof RayTraceResult)
```

Comment:
Hah, I missed that one when converting from PartMOP to the modified version of RayTraceResult, but if that's not a valid change I guess I'll have to revert it back, which is going to be fun...